apps/examples/araplayer/HASParser.cpp

```diff
@@ -0,0 +1,93 @@
+#include <string.h>
+#include "HASParser.h"
+#include "HLSParser.h"
+#include "AudioUtils.h"
+
+HASParser::HASParser()
+{
+	mMapHasParser.insert(std::make_pair(".m3u8", new HLSParser));
+}
+
+HASParserInterface *HASParser::findParserWithPlaylistUrl(const char *url)
+{
+	int urlLen = strlen(url);
+	if (urlLen <= 5) {
+		DEBUG_TRACE(LOG_ERR, "url length <= 5");
+		return NULL;
+	}
+
+	int i;
+	for (i = urlLen - 1; i >= 0 && url[i] != '.'; --i)
+		;
+	if (i < 0) {
+		DEBUG_TRACE(LOG_ERR, "there is no extension");
+		return NULL;
+	}
+
+	ParserIter iter = mMapHasParser.find(url + i);
+	if (iter == mMapHasParser.end()) {
+		DEBUG_TRACE(LOG_ERR, "there is no appropriate parser");
+		return NULL;
+	}
+
+	return iter->second;
+}
+
+int HASParser::parsePlaylist(const char *url, const char *data, int size)
+{
+	HASParserInterface *parser = findParserWithPlaylistUrl(url);
+	if (parser == NULL) {
+		DEBUG_TRACE(LOG_ERR, "parser is null");
+		return -1;
+	}
+
+	return parser->parsePlaylist(url, data, size);
+}
+
+int HASParser::parseSegment(const char *url, const char *data, int size)
+{
+	for (ParserIter iter = mMapHasParser.begin(); iter != mMapHasParser.end(); ++iter) {
+		int ret = iter->second->parseSegment(url, data, size);
```

Comment:
ret is not needed actually.

Reply:
This function is unused, will be deleted in the next commit..